### PR TITLE
Feature/68 improve tests

### LINKED ISSUE #68: Improve tests/coverage
try to understand more deeply how tests work to increase coverage and have real useful and functional tests

### PR DESCRIPTION
**Pull request by jlstoupy98**

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityDescriptionTest.java [hunk 1]
```diff
@@ -29,35 +29,35 @@
...
     @Rule
     public ActivityScenarioRule<ActivityDescriptionActivity> testRule = new ActivityScenarioRule<>(ActivityDescriptionActivity.class);
 
-
-
     @Test
     public void create() {
         Intents.init();
-
+        try{
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityDescriptionTest.java]: Why the thread must sleep before release ?

File: app/src/androidTest/java/com/sdp/movemeet/Backend/BackendActivityManagerDebugTest.java [hunk 2]
```diff
@@ -34,6 +34,9 @@
     @Rule
     public ActivityScenarioRule<BackendActivityManagerDebug> testRule = new ActivityScenarioRule<>(BackendActivityManagerDebug.class);
 
+    @Test
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/Backend/BackendActivityManagerDebugTest.java]: What is this test for ?

File: app/src/androidTest/java/com/sdp/movemeet/map/MapsActivityTest.java [hunk 3]
```diff
@@ -80,6 +81,6 @@ public void mapsActivity_checkThatMarkersAreDisplayedAndClickable() throws UiObj
       / marker1.click();
         marker2.click();
         userMarker.click();
-        // Intents ? */
+        // Intents ?*/
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/map/MapsActivityTest.java]: Comments can be clean if we don't need them

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityDescriptionTest.java (hunk 1 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityDescriptionTest.java]: That was not necessary, I had just forgotten to delete it. Thx !

Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityDescriptionTest.java]: You're welcome

File: app/src/androidTest/java/com/sdp/movemeet/Backend/BackendActivityManagerDebugTest.java (hunk 2 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Backend/BackendActivityManagerDebugTest.java]: Oops, I forgot again to delete it, I will do it now 0:-)

File: app/src/androidTest/java/com/sdp/movemeet/map/MapsActivityTest.java (hunk 3 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/map/MapsActivityTest.java]: That's right, it's a good practice!

File: app/build.gradle [hunk 4]
```diff
@@ -34,6 +34,7 @@ android {
 
     testOptions {
         unitTests.returnDefaultValues = true
+        animationsDisabled = true
```
Comment (Antho1426) [inline app/build.gradle:37]: Good good! Well done to have found this option!

File: app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java [hunk 5]
```diff
@@ -115,6 +119,8 @@ public void onComplete(@NonNull Task<AuthResult> task) {
 
         onView(withId(R.id.editTextStartTime)).perform(forceDoubleClick());
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java:120]: Is forceDoubleClick a trick to force clicking the button for sure?

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+;import java.util.ArrayList;
+import java.util.Date;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+
+@RunWith(AndroidJUnit4.class)
+public class ActivityNotNullDescriptionTest {
```
Comment (bitshrine) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java:26]: What's this test class for?

File: app/src/androidTest/java/com/sdp/movemeet/HomeScreen/HomeScreenActivityTest.java [hunk 7]
```diff
@@ -1,44 +1,34 @@
...
 
     @Rule
-    public ActivityScenarioRule<HomeScreenActivity> testRule = new ActivityScenarioRule<>(HomeScreenActivity.class);
+    public ActivityScenarioRule<HomeScreenActivity> HomeScreenTestRule = new ActivityScenarioRule<>(HomeScreenActivity.class);
 
     @Test
     public void mainActivity_signIn() {
-        Intents.init();
-
         onView(withId(R.id.signInButton)).perform(click());
```
Comment (bitshrine) [inline app/src/androidTest/java/com/sdp/movemeet/HomeScreen/HomeScreenActivityTest.java:26]: We should probably check that the correct Intents are fired on click with an assertion

**Review by Walfar — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Profile/EditProfileActivityTest.java [hunk 8]
```diff
@@ -42,12 +45,6 @@ public void updateProfileByEnteringFieldsAndClickingSaveButton() {
         onView(ViewMatchers.withId(R.id.edit_text_edit_profile_description))
                 .perform(replaceText(TEST_DESCRIPTION), closeSoftKeyboard());
 
-        try {
-            Thread.sleep(1000);
```
Comment (kotleta2007) [inline app/src/androidTest/java/com/sdp/movemeet/Profile/EditProfileActivityTest.java:46]: Why did you remove the Thread.sleep?

**Review by bitshrine — APPROVED** [review]

**Review by kotleta2007 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Profile/EditProfileActivityTest.java (hunk 8 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Profile/EditProfileActivityTest.java:46]: It was a priori useless, the tests passed anyway

Comment (kotleta2007) [reply app/src/androidTest/java/com/sdp/movemeet/Profile/EditProfileActivityTest.java:46]: Nice

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java (hunk 6 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java:26]: This is to test an activity display when the activity is instantiated (via an intent). For the moment, the test just launches the display of the description with a non-null activity. It would be interesting to improve it to check the coherence of the transmitted information

File: app/src/androidTest/java/com/sdp/movemeet/HomeScreen/HomeScreenActivityTest.java (hunk 7 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/HomeScreen/HomeScreenActivityTest.java:26]: Yes, in this case, the intent is created directly when the function is called via the click. We should also check that the click returns the right view.
